reference-architecture/azure-ansible/bastion.sh

```diff
@@ -0,0 +1,345 @@
+master3 openshift_node_labels="{'region':'master','zone':'default'}" 
+node[01:${NODECOUNT}] openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
+infranode1 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+infranode2 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+infranode3 openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
+
+[quotanodes]
+master1 openshift_node_labels="{'region':'master','zone':'default'}" 
+master2 openshift_node_labels="{'region':'master','zone':'default'}" 
+master3 openshift_node_labels="{'region':'master','zone':'default'}" 
+node[01:${NODECOUNT}] openshift_node_labels="{'region': 'primary', 'zone': 'default'}"
+
+[misc]
+store1
+EOF
+
+
+cat <<EOF > /home/${AUSERNAME}/subscribe.yml
+---
+- hosts: all
+  vars:
+    description: "Subscribe OSE"
+  tasks:
+  - name: wait for .updateok
+    wait_for: path=/root/.updateok
+  - name: check connection
+    ping:
+  - name: Get rid of rhui repos
+    file: path=/etc/yum.repos.d/rh-cloud.repo state=absent
+  - name: Get rid of rhui Load balancers
+    file: path=/etc/yum.repos.d/rhui-load-balancers state=absent
+  - name: remove the RHUI package
+    yum: name=RHEL7 state=absent
+  - name: Get rid of old subs
+    shell: subscription-manager unregister
+    ignore_errors: yes
+  - name: register hosts
+    shell: subscription-manager register --username ${RHNUSERNAME} --password ${RHNPASSWORD} 
+    register: task_result
+    until: task_result.rc == 0
+    retries: 10
+    delay: 30
+    ignore_errors: yes
+  - name: attach sub
+    shell: subscription-manager attach --pool=$RHNPOOLID
+    register: task_result
+    until: task_result.rc == 0
+    retries: 10
+    delay: 30
+    ignore_errors: yes
+  - name: disable all repos
+    shell: subscription-manager repos --disable="*" 
+  - name: enable rhel7 repo
+    shell: subscription-manager repos --enable="rhel-7-server-rpms"
+  - name: enable extras repos
+    shell: subscription-manager repos --enable="rhel-7-server-extras-rpms"
+  - name: enable ose repos
+    shell: subscription-manager repos --enable="rhel-7-server-ose-3.3-rpms"
+  - name: install the latest version of PyYAML
+    yum: name=PyYAML state=latest
+  - name: Install the ose client
+    yum: name=atomic-openshift-clients state=latest
+  - name: Update all hosts
+    command: yum -y update
+    async: 1200
+    poll: 10
+  - name: Wait for Things to Settle
+    pause:  minutes=5
+EOF
+
+cat <<EOF > /home/${AUSERNAME}/quota.yml
+---
+- hosts: quotanodes
+  vars:
+    description: "Fix EP Storage/Quota"
+  tasks:
+  - name: Format xfs on disk
+    filesystem: dev=/dev/sdd fstype=xfs force=yes
+  - name: Update Mount to Handle Quota
+    mount: fstype=xfs name=/var/lib/origin/openshift.local/volumes src=/dev/sdd opts="gquota" state="mounted"
+  - name: Change Node perFSGroup Qouta Setting
+    lineinfile: dest=/etc/origin/node/node-config.yaml regexp=^perFSGroup line="    perFSGroup:512Mi"
```

Comment:
cloudinit is a massive work in process, because microsoft "forked" there own, and eventually will converge back, but this is a base evangelistm, and a huge change in ARM as well. 

Also base storage setup really should be done before install, as its part of docker setup, which is a precursor. I do wish I could do different host groups in one playbook, but not seen that one done. Then could potentially combine things. I did find during development, that a bit of separation does improve debuggging, so some spliting was intention to be able to "retry" manually and look at cause of failure.